src/lib/storage/segment_accessor.hpp

```diff
@@ -58,31 +61,53 @@ class SegmentAccessor : public BaseSegmentAccessor<T> {
  * only once.
  */
 template <typename T>
-class MultipleChunkReferenceSegmentAccessor : public BaseSegmentAccessor<T> {
+class MultipleChunkReferenceSegmentAccessor : public AbstractSegmentAccessor<T> {
  public:
-  explicit MultipleChunkReferenceSegmentAccessor(const ReferenceSegment& segment) : _segment{segment} {}
+  static const size_t AVG_CHUNKS = 10;
 
-  const std::optional<T> access(ChunkOffset offset) const final {
-    const auto& referenced_row_id = (*_segment.pos_list())[offset];
-    if (referenced_row_id.is_null()) return std::nullopt;
+  explicit MultipleChunkReferenceSegmentAccessor(const ReferenceSegment& segment)
+      : MultipleChunkReferenceSegmentAccessor(segment, AVG_CHUNKS) {}
 
+  MultipleChunkReferenceSegmentAccessor(const ReferenceSegment& segment, size_t max_chunk_id)
+      : _segment{segment}, _accessors{max_chunk_id + 1} {
     const auto& table = _segment.referenced_table();
-    const auto referenced_column_id = _segment.referenced_column_id();
-    const auto referenced_chunk_id = referenced_row_id.chunk_id;
-    const auto referenced_chunk_offset = referenced_row_id.chunk_offset;
 
-    const auto accessor =
-        create_segment_accessor<T>(table->get_chunk(referenced_chunk_id)->get_segment(referenced_column_id));
-    return accessor->access(referenced_chunk_offset);
+    for (const auto& row : *segment.pos_list()) {
```

Comment:
are we sure preemptively iterating over the entire pos list is a good idea? @mrks, maybe? I feel like lazily checking whether an accessor is available in `access()` might be the faster way?